### PR TITLE
refactor(yt-buying-lib): enable initial space conditions and factor scale in correctly

### PR DESCRIPTION
**Pull request by jparklev**
## Motivation

We got an external request for this library, so I wanted to make sure it was flexible and correct

## Solution

Add the new features from the version of this library in the frontend while maintaining the previous ability to bootstrap the fluxer without a live space pool for data. In addition, add the optimize function into the fluxer directly (rather than assuming the user will bring their own optimization lib)

### CODE REVIEW

File: yt-buying-lib/index.js [hunk 1]
```diff
@@ -40,22 +46,28 @@ class SpaceFluxer {
     return targetOut;
   }
 
-  buyYTs(initialTarget, desiredTargetBack = new Decimal("0")) {
+  // Create a function to a YT buy and return the amount of Target remaining at the end
```
Comment (alexfertel) [inline yt-buying-lib/index.js]:
```suggestion
  // Create a function to make/do a YT buy and return the amount of Target remaining at the end
```

**Review by alexfertel — APPROVED** [review]
LGTM! Now I remembered that I also refactored a bit on the FE side, so thank you for doing this 🙏🏻

File: yt-buying-lib/index.js (hunk 1 above)
Comment (alexfertel) [reply yt-buying-lib/index.js]: Didn't know if the correct word was `make` or `do` 😅

**Review by fedealconada — APPROVED** [review]
looks good!

**Pull request merged**